composer/checkpoint/load.py

```diff
@@ -0,0 +1,519 @@
+        precision (str): The precision to cast the model/optim to after loading
+        cpu_offload (bool): Whether to offload the state dict to CPU before loading.
+        load_planner (Optional[LoadPlanner]): The load planner to use for loading the checkpoint.
+        fsdp_config (Optional[Union[FSDPConfig, dict]]): The FSDP config to use for sharding the model and optimizer.
+        device_mesh (Optional[DeviceMesh]): The device mesh to use for loading the checkpoint.
+        seed (Optional[int]): The seed to use for sharding the model and optimizer.
+
+    """
+
+
+def load_checkpoint(
+    load_path: str,
+    load_options: Optional[Union[CheckpointLoadOptions, Dict]],
+    model: Optional[ComposerModel] = None,
+    optim: Optional[Optimizer] = None,
+    state: Optional[State] = None,
+):
+    """
+    Optionally download and load a checkpoint according to the options into specified state.
+
+    Args:
+        load_path (str): The path or uri to the checkpoint to load or symlink to the path/uri. If URI specified then files will be downloaded first.
+        load_options (CheckpointLoadOptions): The options for loading the checkpoint.
+        model (Optional[ComposerModel]): The model to load the checkpoint into. If not provided, the model from the state will be used.
+        optim (Optional[Optimizer]): The optimizer to load the checkpoint into. If not provided, the optimizer from the state will be used.
+        state (Optional[State]): The state to load the checkpoint. If not provided, the model and optimizer must be provided.
+    """
+    if isinstance(load_options, dict):
+        load_options = CheckpointLoadOptions(**load_options)
+    if load_options.load_model:
+        if model is None:
+            if state is None:
+                raise ValueError('Model or state must be provided if loading model checkpoint.')
+            model = state.model
+
+    if load_options.load_optimizer:
+        if optim is None:
+            if state is None:
+                raise ValueError('Optimizer or state must be provided if loading optimizer checkpoint.')
+            optim = state.optimizers[0]
+
+    if load_options.load_resumption_state and state is None:
+        raise ValueError('State must be provided if loading resumption state.')
+
+    load_model_and_optim_together = (load_options.load_optimizer and load_options.load_model
+                                    ) and (load_options.sharded_checkpoint or load_options.shard_as_needed_during_load)
+    if load_model_and_optim_together:
+        load_model_checkpoint(model, load_path=load_path, optimizer=optim, load_options=load_options)
+
+    else:
+        load_model_checkpoint(model, load_path=load_path, load_options=load_options)
+        if load_options.load_optimizer:
+            load_optim_checkpoint(
+                state.model, state.optimizers[0], load_path, sharded_checkpoint=load_options.sharded_checkpoint
```

Comment:
i remember with V2, model and optimizer state are saved in different folder ? Should we use different load path here ? I don't find the code that append the model and optimizer folder name to the load_path inside the `load_model_checkpoint` and  `load_optim_checkpoint`

Reply:
oh good catch. fixed!